providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -1187,3 +1195,33 @@ def free_reserved_port(self) -> None:
         if self.reserved_tcp_socket:
             self.reserved_tcp_socket.close()
             self.reserved_tcp_socket = None
+
+    def _get_iam_db_login(self) -> str:
+        """Get an IAM login for Cloud SQL database."""
+        if not self.cloudsql_connection.login:
+            raise AirflowException("The login parameter needs to be set in connection")
+
+        if self.database_type == "postgres":
+            return self.cloudsql_connection.login.split(".gserviceaccount.com")[0]
+        else:
+            return self.cloudsql_connection.login.split("@")[0]
+
+    def _generate_login_token(self, service_account) -> str:
+        """Generate an IAM login token for Cloud SQL and return the token."""
+        cmd = ["gcloud", "sql", "generate-login-token", f"--impersonate-service-account={service_account}"]
+        self.log.info("Executing command: %s", " ".join(shlex.quote(c) for c in cmd))
+        success_code = 0
```

Comment:
It is a bit weird that you define variable with 0 as a value and then use it once below.

Maybe then it has to be module const? or I would just use it in "if proc.returncode != 0:" and if you want you can leave inline comment, but I think it has to be clear that 0 is successful return code.

WDYT?

Reply:
Make sense I have updated the code

Reply:
:+1: